### PR TITLE
Master imp tx pull lmi

### PR DESCRIPTION
**Pull request by lmignon**
@pedrobaeza Here it is some improvements and fix to the tx-pull script. The script seems to works fine and has been successfully used to push translation from transifex to l10n-belgium https://github.com/OCA/l10n-belgium/pull/26

### CODE REVIEW

File: tools/tx_pull.py [hunk 1]
```diff
@@ -96,148 +95,200 @@
...
+            os.environ.get('TRANSIFEX_ORGANIZATION') or
+            TX_ORG_DEFAULT)
+        self.gh_org = target or self.tx_org
+        # Connect to GitHub
+        self.github = github_login.login()
+        gh_user = self.github.user()
+
+        if not gh_user.email and not email:
+            raise Exception(
+                'Email required to commit to github. Plz provides one on '
```
Comment (pedrobaeza) [inline tools/tx_pull.py]: s/Plz provides/Please provide

File: tools/tx_pull.py [hunk 2]
```diff
@@ -96,148 +95,200 @@
...
-                tx_project = tx_api.project(project_slug).get()
-                projects.append(tx_project)
-            except:
-                print "ERROR: Transifex project slug %s is invalid" % (
-                    project_slug)
-                return
-    else:
-        start = 1
-        temp_projects = []
-        print "Getting Transifex projects..."
```
Comment (pedrobaeza) [inline tools/tx_pull.py:233]: Don't remove this auto-discovering organization projects part (it's needed for OCA).

**Comment by pedrobaeza** [comment]
One minor remark, and a big one needed for the OCA flow, but overall looks good

File: tools/tx_pull.py (hunk 2 above)
Comment (lmignon) [inline tools/tx_pull.py:233]: @pedrobaeza The auto-discovering has been moved in  `TransifexPuller.process_projects`
https://github.com/acsone/maintainers-tools/blob/master-imp_tx_pull-lmi/tools/tx_pull.py#L199

Comment (pedrobaeza) [inline tools/tx_pull.py:233]: Right, the changes are not easy to follow...

**Comment by pedrobaeza** [comment]
I merge now

**Comment by lmignon** [comment]
@pedrobaeza Thank you!

**Comment by pedrobaeza** [comment]
To you for making the changes

**Pull request merged**